packages/client/src/Call.ts

```diff
@@ -1165,6 +1165,34 @@ export class Call {
     });
   };
 
+  /**
+   * Updates the audio output level for a given participant or for all the participants and the default call audio output level.
+   * @param level
+   * @param participant
+   */
+  setAudioOutputLevel(level: number, participant?: StreamVideoParticipant) {
+    if (participant) {
+      this.state.updateParticipant(participant.sessionId, {
+        audioOutputLevel: level,
+      });
+    } else if (this.sfuClient?.sessionId) {
+      this.state.updateParticipants(
+        this.state.participants.reduce<StreamVideoParticipantPatches>(
+          (acc, p) => {
+            acc[p.sessionId] = {
+              audioOutputLevel: level,
+            };
+            return acc;
+          },
+          {},
+        ),
+      );
+    }
```

Comment:
I don't think you need to maintain integrity between `call.state.audioOutputLevel` and `participant.audioOutputLevel` as this would be prone to errors as we must ensure the level is set whenever a participant is added to the list of participants (see the events).

Instead, `<ParticipantView />` component and do this:
```ts
const defaultAudioOutputLevel = useDefaultAudioOutputLevel();
const audioLevel = participant?.audioOutputLevel ?? defaultAudioOutputLevel;

return (
   <Audio volume={audioLevel} />
   ...
)
```

This way, all changes within the event handlers can be reverted and many edge cases avoided.

Reply:
I think it can be changed at any time, but we just don't have the slider added to the default Settings panel. This isn't a lobby-exclusive feature, as far as I understand it.

Reply:
Would you like me to expose a method on a Call class that would update the default audio output level? Currently the implementation of `Call.setAudioOutputLevel()` is done as the following three exclusive cases:

1. update the single participants level if sessionId
2. update all participants to given level if no sessionId but have joined a call
3. update the call default level.

Reply:
I think it is just enough if we remove the 2nd item. We don't need to keep the audio output level for all participants but only for the ones that had it updated. The SDK's components should do this:
`const volume = p.volume ?? call.volume ?? 1`;

Reply:
I have removed the condition that lead to updating all the participants audio output level.